pkg/api/validate/validators.go

```diff
@@ -247,8 +247,7 @@ func validateAgentPoolProfile(app *api.AgentPoolProfile, vnet *net.IPNet) (errs
 		if app.Name != string(app.Role) {
 			errs = append(errs, fmt.Errorf("invalid properties.agentPoolProfiles[%q]", app.Name))
 		}
-		// TODO: remove count "2" check when CLI is adjusted
-		if app.Count < 2 || app.Count > 3 {
+		if app.Count != 3 {
```

Comment:
Is there a chance large clusters would want more?

Reply:
I guess there is a breaking point for the routers or Azure VMs somewhere. I think that's pretty far away for now, but worth considering as a feature. The size of the infra scale set would have to be customer controlled for that to work at scale, so it's not a small feature.

Reply:
Given that our max supported cluster size is 20 compute nodes right now, I think we don't need to worry about this yet.